### PR TITLE
ADL Formatting

### PR DESCRIPTION
**Pull request by LibbaLawrence**
Have ADL formatting for requests and responses.

### CODE REVIEW

File: packages/python-packages/protocol-stub-generator/protocol/protocol_models.py [hunk 1]
```diff
@@ -671,6 +794,398 @@ def to_json(self):
...
+                    else:
+                        json_request[i] = " : " + param[0] + "[];"
+                if inner_model:
+                    inner_model.append(Token(json_request[i], TokenKind.Comment))
+                    inner_model.append(Token(" ", TokenKind.Newline))
+                else:
+                    if json_request[i] == "str":
+                        json_request[i] = "string"
+                    if json_request[i] == "any-object":
+                        json_request[i] = "{" + "}"
```
Comment (kristapratico) [inline packages/python-packages/protocol-stub-generator/protocol/protocol_models.py:825]: could be refactored into a helper method since you have the same code above

File: packages/python-packages/protocol-stub-generator/protocol/protocol_models.py [hunk 2]
```diff
@@ -671,6 +794,398 @@ def to_json(self):
...
+                            inner_model.append(Token(i, TokenKind.Comment))
+                        else:
+                            inner_model.append(Token(i + ": {", TokenKind.Comment))
+                        name = i
+                    else:
+                        self.add_new_line()
+                        if isinstance(json_request[i], list) and not isinstance(
+                            json_request[i][0], dict
+                        ):
+                            pass
```
Comment (kristapratico) [inline packages/python-packages/protocol-stub-generator/protocol/protocol_models.py:929]: why not just construct the if statement so that you don't need the else below?

File: packages/python-packages/protocol-stub-generator/protocol/protocol_models.py [hunk 3]
```diff
@@ -671,6 +794,398 @@ def to_json(self):
...
+                        else:
+                            self.add_whitespace(indent)
+                            self.add_comment(None, i + ": {", None)  # + ": {"
+                        name = i
+                        inner_model = []
```
Comment (kristapratico) [inline packages/python-packages/protocol-stub-generator/protocol/protocol_models.py:934]: I feel like I've seen `inner_model` defined a few times in this function. Is this to clear the list?

File: packages/python-packages/protocol-stub-generator/protocol/protocol_models.py [hunk 4]
```diff
@@ -671,6 +794,398 @@ def to_json(self):
         return obj_dict
 
 
+def request_builder(
```
Comment (kristapratico) [inline packages/python-packages/protocol-stub-generator/protocol/protocol_models.py:797]: there's a lot of code in this function body. I think we need to refactor out parts so that it's easier to follow.

File: packages/python-packages/protocol-stub-generator/protocol/protocol_models.py [hunk 5]
```diff
@@ -671,6 +794,398 @@ def to_json(self):
...
+                                    TokenKind.Comment,
+                                )
+                            )
+                        else:
+                            inner_model.append(
+                                Token(
+                                    key + ": Map<string, " + param[0] + ">;",
+                                    TokenKind.Comment,
+                                )
+                            )
```
Comment (kristapratico) [inline packages/python-packages/protocol-stub-generator/protocol/protocol_models.py:972]: we can simplify this to have fewer lines of code. Something like:

```suggestion
                        optional = ""
                        if index != -1:
                            optional = "? "
                        inner_model.append(
                            Token(
                                key + optional + ": Map<string, " + param[0] + ">;",
                                TokenKind.Comment,
                            )
                        )
```

comment applies generally

File: packages/python-packages/protocol-stub-generator/protocol/protocol_models.py [hunk 6]
```diff
@@ -671,6 +794,398 @@ def to_json(self):
...
+                                self.add_new_line()
+                                self.add_whitespace(4)
+                                self.add_comment(None, "};", None)
+                                in_model = False
+                                indent = 4
+                    else:
+                        if json_request[i] == "str":
+                            json_request[i] = "string"
+                        if json_request[i] == "any-object":
+                            json_request[i] = "{" + "}"
```
Comment (kristapratico) [inline packages/python-packages/protocol-stub-generator/protocol/protocol_models.py:1015]: I've seen these same lines of code a few times. Convinced they should be refactored out :)

File: packages/python-packages/protocol-stub-generator/protocol/protocol_models.py [hunk 7]
```diff
@@ -702,21 +1217,21 @@ def from_yaml(cls, yaml_data: Dict[str, Any], i, name):
...
+            #     )
+            # else:
+            param_type = get_type(yaml_data["signatureParameters"][i]["schema"])
+            # if param_name == "body":
+            #     try:
+            #         param_name = yaml_data["signatureParameters"][i]["schema"][
+            #             "properties"
+            #         ][0]["serializedName"]
+            #     except:
+            #         param_name = param_name
```
Comment (kristapratico) [inline packages/python-packages/protocol-stub-generator/protocol/protocol_models.py:1234]: remove commented code if not using

**Review by kristapratico — COMMENTED** [review]
Main feedback is to refactor out some of the code from the larger methods into helper functions. It's currently a little hard to follow the indentation and if/else logic when the method spans larger than a screen length.

File: packages/python-packages/protocol-stub-generator/protocol/protocol_models.py [hunk 8]
```diff
@@ -482,6 +517,31 @@ def from_yaml(cls, yaml_data: Dict[str, Any], op_group_num, op_num, namespace):
                         namespace,
                     )
                 )
+                if (
```
[the remark this answers is not in the document]
Comment (LibbaLawrence) [reply packages/python-packages/protocol-stub-generator/protocol/protocol_models.py:520]: I will do that!

File: packages/python-packages/protocol-stub-generator/protocol/protocol_models.py (hunk 3 above)
Comment (LibbaLawrence) [reply packages/python-packages/protocol-stub-generator/protocol/protocol_models.py:933]: yes!

Comment (LibbaLawrence) [reply packages/python-packages/protocol-stub-generator/protocol/protocol_models.py:927]: changed to clear{

File: packages/python-packages/protocol-stub-generator/protocol/protocol_models.py [hunk 9]
```diff
@@ -671,6 +829,385 @@ def to_json(self):
...
+                isinstance(json_request[i], list) or isinstance(json_request[i], dict)
+            ):
+                if i == "str":
+                    if inner_model:
+                        inner_model.append(Token(" ", TokenKind.Newline))
+                        inner_model.append(
+                            Token(" " * (indent * 4), TokenKind.Whitespace)
+                        )
+                        m_type, key = get_map_type(yaml, name)
+                        m_type = format_map_type(m_type)
```
Comment (kristapratico) [inline packages/python-packages/protocol-stub-generator/protocol/protocol_models.py:871]: it doesn't seem like `m_type` is used after being defined

File: packages/python-packages/protocol-stub-generator/protocol/protocol_models.py [hunk 10]
```diff
@@ -671,6 +829,385 @@ def to_json(self):
...
+                    if not key: key = self.parameters[0].name
+                    if inner_model:
+                        if index != -1:
+                            optional = "?"
+                        inner_model.append(
+                            Token(
+                                key + optional + " : Map<string, " + param[0] + ">;",
+                                TokenKind.Comment,
+                            )
+                        )
```
Comment (kristapratico) [inline packages/python-packages/protocol-stub-generator/protocol/protocol_models.py:958]: Need to define optional in case the `if index != -1` statement isn't satisfied since it will be used in the append() call.

```suggestion
                        optional = ""
                        if index != -1:
                            optional = "?"
                        inner_model.append(
                            Token(
                                key + optional + " : Map<string, " + param[0] + ">;",
                                TokenKind.Comment,
                            )
                        )
```

File: packages/python-packages/protocol-stub-generator/protocol/protocol_models.py [hunk 11]
```diff
@@ -671,6 +829,385 @@ def to_json(self):
...
+                    name=name,
+                    inner_model=inner_model,
+                    no_list=no_list,
+                )
+                inner_model = format_punctuation(self, json_request, indent, inner_model, i)
+    return not_first,indent,inner_model
+
+def format_punctuation(self, json_request, indent, inner_model, i):
+    if i == "str":
+        pass
```
Comment (kristapratico) [inline packages/python-packages/protocol-stub-generator/protocol/protocol_models.py:1025]: instead of `pass` you could just put the return statement here

File: packages/python-packages/protocol-stub-generator/protocol/protocol_models.py [hunk 12]
```diff
@@ -671,6 +829,385 @@ def to_json(self):
...
+                            Token(" " * (indent * 4), TokenKind.Whitespace)
+                        )
+            inner_model.append(Token("};", TokenKind.Comment))
+        else:
+            self.add_new_line()
+            self.add_whitespace(indent)
+            self.add_comment(None, "};", None)
+    return inner_model
+
+def list_format(self, json_request, yaml, not_first, indent, name, inner_model):
```
Comment (kristapratico) [inline packages/python-packages/protocol-stub-generator/protocol/protocol_models.py:1079]: The value for the `not_first` param doesn't get used in the function body -- not sure if intentional.

File: packages/python-packages/protocol-stub-generator/protocol/protocol_models.py [hunk 13]
```diff
@@ -671,6 +829,385 @@ def to_json(self):
...
+    if isinstance(json_request, list):
+        for i in range(0, len(json_request)):
+            if isinstance(json_request[i], str):
+                index = json_request[i].find("(optional)")
+                param = json_request[i].split()
+                if len(param) >= 2:
+                    param[0] = format_param_type(param[0])
+                    if index != -1:
+                        optional = "?"
+                    json_request[i] = optional + " : " + param[0] + "[];"
```
Comment (kristapratico) [inline packages/python-packages/protocol-stub-generator/protocol/protocol_models.py:1090]: same comment for `optional` here, make sure it's defined as an empty string before the if statement otherwise you could hit a `Local variable 'optional' might be referenced before assignment`

**Review by kristapratico — COMMENTED** [review]
Looking good! Two things:

1) You can remove all the files that got committed under `__pycache__`
2) For the autorest vendored code, I believe you only use the code under [autorest/codegen/models](https://github.com/Azure/autorest.python/tree/autorestv3/autorest/codegen/models). Can you check that and remove the files that you aren't using? thank you~

File: packages/python-packages/protocol-stub-generator/protocol/protocol_models.py (hunk 9 above)
Comment (LibbaLawrence) [reply packages/python-packages/protocol-stub-generator/protocol/protocol_models.py:868]: it's used in inner_model

**Review by kristapratico — APPROVED** [review]
